metagraph-plugin/metagraph_katana/plugins/metagraph_katana/translators.py

```diff
@@ -0,0 +1,58 @@
+import metagraph as mg
+import networkx as nx
+import numpy as np
+import pyarrow as pa
+from metagraph import translator
+from metagraph.plugins.networkx.types import NetworkXGraph
+from scipy.sparse import csr_matrix
+
+from katana.property_graph import PropertyGraph
+
+from .types import KatanaGraph
+
+
+@translator
+def networkx_to_katanagraph(x: NetworkXGraph, **props) -> KatanaGraph:
+    aprops = NetworkXGraph.Type.compute_abstract_properties(x, {"node_dtype", "node_type", "edge_type", "is_directed"})
+    is_weighted = aprops["edge_type"] == "map"
+    # get the edge list directly from the NetworkX Graph
+    elist_raw = list(x.value.edges(data=True))
+    # sort the eddge list and node list
+    if aprops["is_directed"]:
+        elist = sorted(elist_raw, key=lambda each: (each[0], each[1]))
+    else:
+        inv_elist = [(each[1], each[0], each[2]) for each in elist_raw]
+        elist = sorted(elist_raw + inv_elist, key=lambda each: (each[0], each[1]))
```

Comment:
Doesn't this permute the edges and nodes? How is that captured so the user can map the nodes and edges back to the original if they want. If we don't support that, we should make sure to document that.